### ISSUE TITLE
Remove all references to python 2

### ISSUE THREAD

**Issue by nden** [spacetelescope/jwst#1392]
This will be done possibly in a sprint. The ticket here is to collect comments and some obvious things we need to look for. 

- Remove all `from __future__ import ...`
- Remove six
- Remove any direct references to python versions
- Classes don't need to inherit from `object` any more
- Remove `unicode_literals`, if there are any

subpackages
- [x] ami
- [x] assign_wcs
- [x] associations
- [x] background
- [x] barshadow
- [x] combine_1d
- [x] coron
- [x] csv_tools
- [x] cube_build
- [x] cube_skymatch
- [x] dark_current
- [x] datamodels
- [x] dq_init
- [x] emission
- [x] engdblog
- [x] exp_to_source
- [x] extract_1d
- [x] extract_2d
- [x] firstframe
- [x] fits_generator
- [x] flatfield
- [x] fringe
- [x] gain_scale
- [x] group_scale
- [x] guider_cds
- [x] imprint
- [x] ipc
- [x] jump
- [x] jwpsf
- [x] lastframe
- [x] lib
- [x] linearity
- [x] model_blender
- [x] mrs_imatch
- [x] msaflagopen
- [x] outlier_detection
- [x] pathloss
- [x] persistence
- [x] photom
- [x] pipeline
- [x] ramp_fitting
- [x] refpix
- [x] resample
- [x] reset
- [x] rscd
- [x] saturation
- [x] skymatch
- [x] source_catalog
- [x] srctype
- [x] stpipe
- [x] straylight
- [x] superbias
- [x] tests
- [x] timeconversion
- [x] transforms
- [x] tso_photometry
- [x] tweakreg
- [x] tweakreg_catalog
- [x] wfs_combine
- [x] white_light
- [x] wiimatch

**Comment by nden** [comment]
@STScI-JWST/maintainers Here's a list of all subdirectories. When you update a step please check it out so we all have an idea how this is progressing.
(Although I think it's probably best to do it approximately at the same time.)

**Comment by nden** [comment]
@STScI-JWST/maintainers Please don't forget to update the table in this issue. To facilitate finding it, there's a link now to it under `projects`.

**Comment by hbushouse** [comment]
@nden There are still a couple of references to "six" in ``transforms/jwst_types.py``.

**Comment by nden** [comment]
@hbushouse Yeah, let's keep these for now. I'll do something about them in a separate PR.

**Comment by stscirij** [comment]
fits_generator updates to remove Python 2 were handled in #2382

**Comment by hbushouse** [comment]
So ... can this be closed?

**Comment by stscirij** [comment]
Not quite.  I have a few classes that inherit from 'object' that I have a PR open for (#2678).  There's also a few more such cases in datamodels and lib/signal_slot

**Comment by stscirij** [comment]
I added them to the PR #2678, which should take care of everything.  Once that PR is merged, we can close this.
